tree/tree/src/TTree.cxx

```diff
@@ -6580,7 +6580,12 @@ void TTree::OptimizeBaskets(ULong64_t maxMemory, Float_t minComp, Option_t *opti
          if (bsize < 0) bsize = bmax;
          if (bsize > bmax) bsize = bmax;
          UInt_t newBsize = UInt_t(bsize);
-         newBsize = newBsize - newBsize%512;
+         if (pass) { // only on the second pass so that it doesn't interfere with scaling
+            if (branch->GetBasket(0) != 0) {
+               newBsize = newBsize + (branch->GetBasket(0)->GetNevBuf() * sizeof(Int_t) * 2); // make room for meta dat
+            }
+            newBsize = newBsize - newBsize%512 + 512; // rounds up
```

Comment:
This changes the minimum size in case of small basket to at least 512.  Do we know the consequence in case of unbalanced trees (a few very large branches and many small branches)?